test/cases/api.sh

```diff
@@ -613,6 +665,46 @@ function _instanceWaitSSH() {
   done
 }
 
+function _instanceCheck() {
+  echo "✔️ Instance checking"
+  local _ssh="$1"
+
+  # Check if postgres is installed
+  $_ssh rpm -q postgresql
+
+  pretty_name=$($_ssh grep '^PRETTY_NAME=' /etc/os-release)
+  # Beta image doesn't have available subscription so skip subscription check. See COMPOSER-996
+  if [[ "$ID" =~ "rhel" ]] && [[ ! "$pretty_name" =~ 'Beta' ]]; then
+    # Verify subscribe status. Loop check since the system may not be registered such early
+    set +eu
+    for LOOP_COUNTER in {1..10}; do
+        subscribe_org_id=$($_ssh sudo subscription-manager identity | grep 'org ID')
+        if [[ "$subscribe_org_id" == "org ID: $API_TEST_SUBSCRIPTION_ORG_ID" ]]; then
+            echo "System is subscribed."
+            break
+        else
+            echo "System is not subscribed. Retrying in 30 seconds...($LOOP_COUNTER/10)"
+            sleep 30
+        fi
+    done
+    set -eu
+    [[ "$subscribe_org_id" == "org ID: $API_TEST_SUBSCRIPTION_ORG_ID" ]]
+
+    # Remove all repos except the RHSM repo
+    $_ssh sudo find /etc/yum.repos.d/ ! -name 'redhat.repo' -type f -exec rm -f {} +
+    # Clean cache and get enabled repo list
+    $_ssh sudo rm -rf /var/cache/dnf
+    # Verify package installation. Disable *beta* repos because they are not reachable
+    $_ssh sudo dnf repolist
```

Comment:
B/c the origins of this code-block were copied from the image-builder repository and b/c the `rpm -q dos2unix` check was removed in https://github.com/osbuild/image-builder/pull/190#discussion_r644010182 and b/c this PR is still open IMO it makes sense to just add the check back. This is equivalent to fixing https://github.com/osbuild/image-builder/issues/195 inside the osbuild-composer repository.

Reply:
Hi Alex,
I've removed all the other repos before installation. And due to https://github.com/osbuild/image-builder/pull/190#discussion_r644002358 we have confirmed that if there's no available repo, it will returns 1 no matter whether this package is installed or not. So that I think there's no need to add the pre-check anymore. Is that acceptable? Thanks!

Reply:
Show me where the repos are removed b/c I am not seeing it.

Reply:
That's a hell of a weird way to make sure your pre-condition is met. Let's hope dnf doesn't change behavior.